plugins/plugin-maven/che-plugin-maven-ide/src/main/java/org/eclipse/che/plugin/maven/client/actions/GetEffectivePomAction.java

```diff
@@ -77,49 +80,49 @@ public void updateInPerspective(@NotNull ActionEvent event) {
       return;
     }
 
-    final Optional<Project> project = resource.getRelatedProject();
-    if (!project.isPresent()) {
+    final Project project = resource.getProject();
+    if (project == null) {
       event.getPresentation().setEnabledAndVisible(false);
       return;
     }
 
-    event.getPresentation().setEnabledAndVisible(project.get().isTypeOf(MAVEN_ID));
+    event.getPresentation().setEnabledAndVisible(project.isTypeOf(MAVEN_ID));
   }
 
   @Override
   public void actionPerformed(ActionEvent e) {
-    final Resource[] resources = appContext.getResources();
-
-    checkState(resources != null && resources.length == 1);
-
-    final Project project = resources[0].getRelatedProject().get();
+    final Resource resource = appContext.getResource();
+    checkState(resource != null);
 
+    final Project project = resource.getProject();
     checkState(MAVEN_ID.equals(project.getType()));
 
-    mavenServerServiceClient
-        .getEffectivePom(project.getLocation().toString())
+    final String pathToProjectsDir = "/projects";
+    final String absolutePathToProjectPom =
+        pathToProjectsDir + project.getLocation().toString() + "/pom.xml";
+    GetEffectivePomParameters paramsDto =
+        dtoFactory
+            .createDto(GetEffectivePomParameters.class)
+            .withPathToProjectPom(absolutePathToProjectPom);
+
+    javaLanguageExtensionServiceClient
+        .effectivePom(paramsDto)
         .then(
-            new Operation<String>() {
-              @Override
-              public void apply(String content) throws OperationException {
-                editorAgent.openEditor(
-                    new SyntheticFile(
-                        "pom.xml",
-                        project.getAttributes().get(MavenAttributes.ARTIFACT_ID).get(0)
-                            + " [effective pom]",
-                        content));
-              }
+            content -> {
+              editorAgent.openEditor(
+                  new SyntheticFile(
+                      "pom.xml",
```

Comment:
Can you test what happens when you open two editors with effective poms? "SyntheticFile" uses the passed in name as the path, this might mean the system thinks the file is already open.

Reply:
Yes, good catch, sometimes it could cause problems.